noir_stdlib/src/sha256.nr

```diff
@@ -40,21 +40,21 @@ fn hash_final_block(msg_block: [u8; 64], mut state: [u32; 8]) -> [u8; 32] {
 }
 
 // Variable size SHA-256 hash
+#[no_predicates]
 pub fn sha256_var<let N: u32>(msg: [u8; N], message_size: u64) -> [u8; 32] {
     let mut msg_block: [u8; 64] = [0; 64];
     let mut h: [u32; 8] = [1779033703, 3144134277, 1013904242, 2773480762, 1359893119, 2600822924, 528734635, 1541459225]; // Intermediate hash, starting with the canonical initial value
     let mut i: u64 = 0; // Message byte pointer
+    assert(message_size <= N as u64);
```

Comment:
This breaks the rules on `no_predicates` functions right?

Reply:
Ah yes you are right. I was going to put it in a wrapper but I am realizing do we even need the assert? `message_size` is only used at the end for being attached to the final block.